docs/form-guide.md

```diff
@@ -0,0 +1,108 @@
+# Form Guide
+
+## General content
+
+We use [`react-hook-form`](https://react-hook-form.com/) and [`zod`](https://github.com/colinhacks/zod#error-formatting) to create the form. Here are the basic steps to create a simple form
+
+### 1. Use `zod` to create form field schema
+
+```js
+import { z as zod } from 'zod'
+
+const schema = zod.object({
+  name: zod.string(), // string
+  age: zod.number('The age need be a number').positive('The age need to greater than 0'), // > 0
+  country: zod.string('The country need be a string').optional(), // string | undefiend
+  address: zod
+    .string()
+    .min(1)
+    .refine((address) => ValidAddress(address), 'Invalid Address'), // You can use other methods to validate this field
+})
+```
+
+### 2. Create type to be compatible with schema
+
+```
+type formType = {
```

Comment:
This is a huge waste of the `zod` library ability. `zod` support inference type from the schema object. See https://www.npmjs.com/package/zod#type-inference